src/terrain/generation/noise.hpp

```diff
@@ -24,7 +24,10 @@ namespace generation {
 
 class Noise {
```

Comment:
Shouldn't this have some virtual methods or something?

Reply:
It wouldn't heart, but I don't want to use virtual classes with sol/lua. More clear to do that explicitly.

Reply:
But then subclassing doesn't make sense

Reply:
1) If I want to put noise into a function template

2) All Noise derived classes need get double

3) I'm fine using virtual functions, just not with lua

Reply:
1. But it doesn't have any public methods.
2. Doesn't need to be in a class though. Its static
3. But a class with no public methods and no friends is useless.

Reply:
1) That's on me. It should have virtuial double get_noise(double, double);
2) I could make it a static function, but I want less things floating around.
3) get_double is used in three other classes. And I think I will use this as a concept.

Reply:
1. Add that then
2. A static function can be defined in a single cpp file.
3. Then make a `concept Noise` and throw out the class. A class and a concept are two very different things.

Reply:
Im not sure what you are sujesting here. Get double is used in two files already. Everything written is relatvly clean. I don't see a reason to change anything.